SimTKcommon/include/SimTKcommon/internal/Pathname.h

```diff
@@ -87,6 +87,29 @@ namespace SimTK {
  */
 class SimTK_SimTKCOMMON_EXPORT Pathname {
 public:
+    /// Given a specified working directory (swd) and path, this function
+    /// evaluates the absolute path of a given path relative to the swd and
+    /// returns the directory, fileName, and extension of the canonicalized path
+    /// with respect to a swd, if needed. This means, for the path, that instead of 
+    /// evaluating "." as the current working directory (cwd), the swd is used. This
+    /// function also returns a boolean (dontApplySearchPath) to indicate if a search
+    /// path should not be applied. The following rules are applied in order:
```

Comment:
This should be more clear about the meaning of this flag. Something like "... to indicate whether it would be appropriate to apply a search path if this is being used in a search path context, like command look up in a shell. If a specific directory was specified in some way then a search should not be done. For example `mypath` or `mypath/myfile` could be searched, but `./mypath` could not."